src/p2p/Sync.ts

```diff
@@ -551,154 +552,171 @@ export function totalNodeCount(cycle: P2P.CycleCreatorTypes.CycleRecord) {
-      refuted: 'a',
-      joined: 'a',
-      returned: 'a',
-      apoptosized: 'a',
-      networkDataHash: 'a',
-      networkReceiptHash: 'a',
-      networkSummaryHash: 'a',
-      desired: 'n',
-    })
-    if (err) {
-      warn('Type validation failed for cycleRecord: ' + err)
-    }
-    for (const refreshedArchiver of cycleRecord.refreshedArchivers) {
-      err = validateTypes(refreshedArchiver, archiverType)
-      if (err) {
-        warn('Validation failed for cycleRecord.refreshedArchivers: ' + err)
-        return false
-      }
-    }
-    for (const refreshedConsensor of cycleRecord.refreshedConsensors) {
-      err = validateTypes(refreshedConsensor, {
-        curvePublicKey: 's',
-        status: 's',
-      })
-      if (err) {
-        warn('Validation failed for cycleRecord.refreshedConsensors: ' + err)
-        return false
-      }
-    }
-    for (const joinedArchiver of cycleRecord.joinedArchivers) {
-      err = validateTypes(joinedArchiver, archiverType)
-      if (err) {
-        warn('Validation failed for cycleRecord.joinedArchivers: ' + err)
-        return false
-      }
-    }
-    for (const leavingArchiver of cycleRecord.leavingArchivers) {
-      err = validateTypes(leavingArchiver, archiverType)
-      if (err) {
-        warn('Validation failed for cycleRecord.leavingArchivers: ' + err)
-        return false
-      }
-    }
-    for (const joinedConsensor of cycleRecord.joinedConsensors) {
-      err = validateTypes(joinedConsensor, {
-        cycleJoined: 's',
-        counterRefreshed: 'n',
-        id: 's',
-      })
-      if (err) {
-        warn('Validation failed for cycleRecord.joinedConsensors: ' + err)
-        return false
-      }
-    }
-    for (const nodeId of cycleRecord.activated) {
-      if (typeof nodeId !== 'string') {
-        warn('Validation failed for cycleRecord.refreshedConsensors: ' + err)
-        return false
-      }
-    }
-    for (const activatedPublicKey of cycleRecord.activatedPublicKeys) {
-      if (typeof activatedPublicKey !== 'string') {
-        warn('Validation failed for cycleRecord.activatedPublicKeys: ' + err)
-        return false
-      }
-    }
-    for (const nodeId of cycleRecord.lost) {
-      if (typeof nodeId !== 'string') {
-        warn('Validation failed for cycleRecord.lost: ' + err)
-        return false
-      }
-    }
-    for (const nodeId of cycleRecord.refuted) {
-      if (typeof nodeId !== 'string') {
-        warn('Validation failed for cycleRecord.refuted: ' + err)
-        return false
-      }
-    }
-    for (const nodeId of cycleRecord.joined) {
-      if (typeof nodeId !== 'string') {
-        warn('Validation failed for cycleRecord.joined: ' + err)
-        return false
-      }
-    }
-    for (const nodeId of cycleRecord.returned) {
-      if (typeof nodeId !== 'string') {
-        warn('Validation failed for cycleRecord.returned: ' + err)
-        return false
-      }
+  if (config.p2p.useAjvCycleRecordValidation) {
```

Comment:
I've talked with Andrew about this. The reason for the toggle is cycleRecord validation is a central part and in case we are too strict with our schema (e.g. some weird edge case) that would cause cycle records to not become valid anymore we will want to switch back. 
This should be phased out at a later point when we are more confident with this (can be done with RED-300 - should add a test by taking the cycle records from a long running network ideally ITN3)